TFNetworkRecLayer.py

```diff
@@ -862,6 +862,16 @@ def __call__(lself, name):
           if construct_ctx.layers:
             construct_ctx.layers[-1].dependencies.add(layer)
           return layer
+        if '/' in name:
+          # this is probably a path to a sub-layer
+          root_name = name.split('/')[0]
+          root_layer = lself.__call__(root_name)  # get the root-layer (first part of the path)
+          sub_layer = root_layer.get_sub_layer('/'.join(name.split('/')[1:]))  # get the sub-layer from the root-layer
+          if sub_layer:  # get_sub_layer returns None by default (if sub-layer not found)
+            # add to templates so we will collect output in self.get_output if this is an output layer
+            self.layer_data_templates[name] = sub_layer
+            sub_layer.dependencies.add(root_layer)
+            return sub_layer
```

Comment:
The logic for sub layers inside the rec layer template construction, do you really need that? If not, I would vote for removing this, to keep the code simpler for now. But at least it should be a separate pull request.

Reply:
Ah I see, this is needed for the ChoiceLayer which provides the different sub layers.